### PR TITLE
ENH boost 1.70 w/ icu 67

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR is intended for a 1.70.x branch on master.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by isuruf** [comment]
Please add yourself as a maintainer.

File: recipe/build.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ CXXFLAGS="${CXXFLAGS} -fPIC"
 if [ "$(uname)" == "Darwin" ]; then
     TOOLSET=clang
 elif [ "$(uname)" == "Linux" ]; then
-    TOOLSET=gcc
+    TOOLSET=cc
```
Comment (beckermr) [inline recipe/build.sh:22]: This change was in the old 1.70 build

**Comment by beckermr** [comment]
@isuruf do you want to pull this to a branch, or shall I?

**Pull request merged**